test-runner-poc/lib/runner.js

```diff
@@ -19,22 +19,27 @@ function addFilter(filter) {
   filters.push(filter);
 
 }
-function environmentSetup(environmentCallback, done) {
+function environmentSetup(environmentCallback) {
   const mongodb_uri = process.env.MONGODB_URI || 'mongodb://localhost:27017';
   mongoClient = new MongoClient(mongodb_uri);
   mongoClient.connect((err, client) => {
-    if (err) environmentCallback(err);
+    if (err) {
+      environmentCallback(err);
+      return;
+    }
     createFilters(environmentParser);
     function environmentParser(environmentName, version) {
       console.log('environmentName ', environmentName);
       const Environment = environments[environmentName];
       let host, port;
       parseConnectionString(mongodb_uri, (err, parsedURI) => {
-        if (err) environmentCallback(err);
+        if (err) {
+          environmentCallback(err);
```

Comment:
This can be `return environmentCallback(err);`

Reply:
FIxed